### PR TITLE
refactor: use code generate for moduledata

### PR DESCRIPTION
**Pull request by Zxilly**
I'd like to use this PR as part of the continue work of #69.

@TcM1911 Can you review this?

### CODE REVIEW

**Comment by Zxilly** [comment]
Closes: #79

**Comment by Zxilly** [comment]
Some of code from https://github.com/goretk/gore/pull/69.

**Comment by Zxilly** [comment]
I'd like to use this PR as part of the continue work of #69.

@TcM1911 Can you review this?

**Comment by TcM1911** [comment]
Very nice! I've wanted to clean up the generation code for a long time. I really appreciate this PR.

There is a "bug" in the generation code for std packages that's been present since the beginning, and maybe you want to tackle it too. The logic will always use the "current" available packages from the Go repo. So when they remove internal packages or move things around, the "old" packages are removed from the list. This causes the package detection for older binaries to get worse with newer versions of Gore. The proper way to handle this would be to append only new packages to the list and not regenerate it every time. We also will have to recover the packages that has been removed and re-add them to the list. This could probably be done from Gore's git history.

**Comment by Zxilly** [comment]
It has been implemented. Plz check the stdpkgs.go
I implement it by searching all the minor release version of golang, then merge them.

**Comment by Zxilly** [comment]
I'm writing an auto generate logic for moduledata. Playing with ast is not easy and may take some hours.

**Comment by Zxilly** [comment]
Now moduledata will auto generate from golang codebase. 

@TcM1911 I think this PR is ready for review.

**Comment by Zxilly** [comment]
Any updates?

**Review by StableAgOH — APPROVED** [review]

**Comment by Zxilly** [comment]
If there any problem blocking the merge process please let me know

**Comment by TcM1911** [comment]
Sorry @Zxilly, I've just been very busy. I will take a look at this today. There are some conflicts, can you fix them?

File: elf.go [hunk 1]
```diff
@@ -45,11 +45,11 @@ type elfFile struct {
 	osFile *os.File
 }
 
-func (e *elfFile) GetParsedFile() any {
+func (e *elfFile) getParsedFile() any {
```
Comment (TcM1911) [inline elf.go:48]: This is reverting #74.

**Comment by Zxilly** [comment]
conflict resolved.

File: elf.go (hunk 1 above)
Comment (Zxilly) [reply elf.go:48]: The public API is still available. Since elf/macho/peFile is not exported, has a public method on it has no meaning.

**Pull request merged**